File: matiasmasso/Mat.NET/DTO/DTORepCliCom.cs
using System;

namespace DTO
{
    public class DTORepCliCom : DTOBaseGuid
    {
        public DTORep Rep { get; set; }
        public DTOCustomer Customer { get; set; }
        public ComCods ComCod { get; set; }
        public DateTime Fch { get; set; }
        public string Obs { get; set; }
        public DTOUser UsrCreated { get; set; }
        public DateTime FchCreated { get; set; }

        public enum ComCods
        {
            Standard,
            Reduced,
            Excluded
        }

        public DTORepCliCom() : base()
        {
        }

        public DTORepCliCom(Guid oGuid) : base(oGuid)
        {
        }

        public static DTORepCom RepCom(DTORepCliCom oRepCliCom, DTORepProduct oRepProduct)
        {
            DTORepCom retval = null/* TODO Change to default(_) if this is not a reference type */;
            switch (oRepCliCom.ComCod)
            {
                case DTORepCliCom.ComCods.Standard:
                    {
                        retval = new DTORepCom();
                        retval.Rep = oRepProduct.Rep;
                        retval.Com = oRepProduct.ComStd;
                        break;
                    }

                case DTORepCliCom.ComCods.Reduced:
                    {
                        retval = new DTORepCom();
                        retval.Rep = oRepProduct.Rep;
                        retval.Com = oRepProduct.ComRed;
                        break;
                    }
            }
            return retval;
        }

        public static decimal Com(DTORepCliCom oRepCliCom, DTORepProduct oRepProduct)
        {
            decimal retval = 0;
            switch (oRepCliCom.ComCod)
            {
                case DTORepCliCom.ComCods.Standard:
                    {
                        retval = oRepProduct.ComStd;
                        break;
                    }

                case DTORepCliCom.ComCods.Reduced:
                    {
                        retval = oRepProduct.ComRed;
                        break;
                    }
            }
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTORepCliCom.vb
Public Class DTORepCliCom
    Inherits DTOBaseGuid

    Property Rep As DTORep
    Property Customer As DTOCustomer
    Property ComCod As ComCods
    Property Fch As Date
    Property Obs As String
    Property UsrCreated As DTOUser
    Property FchCreated As DateTime

    Public Enum ComCods
        Standard
        Reduced
        Excluded
    End Enum

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function RepCom(oRepCliCom As DTORepCliCom, oRepProduct As DTORepProduct) As DTORepCom
        Dim retval As DTORepCom = Nothing
        Select Case oRepCliCom.ComCod
            Case DTORepCliCom.ComCods.Standard
                retval = New DTORepCom
                retval.Rep = oRepProduct.Rep
                retval.Com = oRepProduct.ComStd
            Case DTORepCliCom.ComCods.Reduced
                retval = New DTORepCom
                retval.Rep = oRepProduct.Rep
                retval.Com = oRepProduct.ComRed
        End Select
        Return retval
    End Function

    Shared Function Com(oRepCliCom As DTORepCliCom, oRepProduct As DTORepProduct) As Decimal
        Dim retval As Decimal = 0
        Select Case oRepCliCom.ComCod
            Case DTORepCliCom.ComCods.Standard
                retval = oRepProduct.ComStd
            Case DTORepCliCom.ComCods.Reduced
                retval = oRepProduct.ComRed
        End Select
        Return retval
    End Function
End Class
